File: Vanilla/sources/zomboid/fmod/fmod/FMODAudio.java
package fmod.fmod;

import zombie.audio.BaseSoundEmitter;

public class FMODAudio implements Audio {
   public BaseSoundEmitter emitter;

   public FMODAudio(BaseSoundEmitter var1) {
      this.emitter = var1;
   }

   public boolean isPlaying() {
      return !this.emitter.isEmpty();
   }

   public void setVolume(float var1) {
      this.emitter.setVolumeAll(var1);
   }

   public void start() {
   }

   public void pause() {
   }

   public void stop() {
      this.emitter.stopAll();
   }

   public void setName(String var1) {
   }

   public String getName() {
      return null;
   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/FMODAudio.lua
---@class FMODAudio : fmod.fmod.FMODAudio
---@field public emitter BaseSoundEmitter
FMODAudio = {}

---@public
---@return void
function FMODAudio:stop() end

---@public
---@return boolean
function FMODAudio:isPlaying() end

---@public
---@return void
function FMODAudio:pause() end

---@public
---@return String
function FMODAudio:getName() end

---@public
---@return void
function FMODAudio:start() end

---@public
---@param arg0 float
---@return void
function FMODAudio:setVolume(arg0) end

---@public
---@param arg0 String
---@return void
function FMODAudio:setName(arg0) end
